lib/std/unicode.zig

```diff
@@ -741,3 +746,631 @@ test "utf8ToUtf16LeStringLiteral" {
+        0x16800...0x16A3F => .BamumSupplement,
+        0x16A40...0x16A6F => .Mro,
+        0x16AD0...0x16AFF => .BassaVah,
+        0x16B00...0x16B8F => .PahawhHmong,
+        0x16E40...0x16E9F => .Medefaidrin,
+        0x16F00...0x16F9F => .Miao,
+        0x16FE0...0x16FFF => .IdeographicSymbolsAndPunctuation,
+        0x17000...0x187FF => .Tangut,
+        0x18800...0x18AFF => .TangutComponents,
+        0x1B000...0x1B0FF => .KanaSupplement,
+        0x1B100...0x1B12F => .KanaExtendedA,
+        0x1B130...0x1B16F => .SmallKanaExtension,
+        0x1B170...0x1B2FF => .Nushu,
+        0x1BC00...0x1BC9F => .Duployan,
+        0x1BCA0...0x1BCAF => .ShorthandFormatControls,
+        0x1D000...0x1D0FF => .ByzantineMusicalSymbols,
+        0x1D100...0x1D1FF => .MusicalSymbols,
+        0x1D200...0x1D24F => .AncientGreekMusicalNotation,
+        0x1D2E0...0x1D2FF => .MayanNumerals,
+        0x1D300...0x1D35F => .TaiXuanJingSymbols,
+        0x1D360...0x1D37F => .CountingRodNumerals,
+        0x1D400...0x1D7FF => .MathematicalAlphanumericSymbols,
+        0x1D800...0x1DAAF => .SuttonSignWriting,
+        0x1E000...0x1E02F => .GlagoliticSupplement,
+        0x1E100...0x1E14F => .NyiakengPuachueHmong,
+        0x1E2C0...0x1E2FF => .Wancho,
+        0x1E800...0x1E8DF => .MendeKikakui,
+        0x1E900...0x1E95F => .Adlam,
+        0x1EC70...0x1ECBF => .IndicSiyaqNumbers,
+        0x1ED00...0x1ED4F => .OttomanSiyaqNumbers,
+        0x1EE00...0x1EEFF => .ArabicMathematicalAlphabeticSymbols,
+        0x1F000...0x1F02F => .MahjongTiles,
+        0x1F030...0x1F09F => .DominoTiles,
+        0x1F0A0...0x1F0FF => .PlayingCards,
+        0x1F100...0x1F1FF => .EnclosedAlphanumericSupplement,
+        0x1F200...0x1F2FF => .EnclosedIdeographicSupplement,
+        0x1F300...0x1F5FF => .MiscellaneousSymbolsAndPictographs,
+        0x1F600...0x1F64F => .Emoticons,
+        0x1F650...0x1F67F => .OrnamentalDingbats,
+        0x1F680...0x1F6FF => .TransportAndMapSymbols,
+        0x1F700...0x1F77F => .AlchemicalSymbols,
+        0x1F780...0x1F7FF => .GeometricShapesExtended,
+        0x1F800...0x1F8FF => .SupplementalArrowsC,
+        0x1F900...0x1F9FF => .SupplementalSymbolsAndPictographs,
+        0x1FA00...0x1FA6F => .ChessSymbols,
+        0x1FA70...0x1FAFF => .SymbolsAndPictographsExtendedA,
+        0x20000...0x2A6DF => .CJKUnifiedIdeographsExtensionB,
+        0x2A700...0x2B73F => .CJKUnifiedIdeographsExtensionC,
+        0x2B740...0x2B81F => .CJKUnifiedIdeographsExtensionD,
+        0x2B820...0x2CEAF => .CJKUnifiedIdeographsExtensionE,
+        0x2CEB0...0x2EBEF => .CJKUnifiedIdeographsExtensionF,
+        0x2F800...0x2FA1F => .CJKCompatibilityIdeographsSupplement,
+        0xE0000...0xE007F => .Tags,
+        0xE0100...0xE01EF => .VariationSelectorsSupplement,
+        0xF0000...0xFFFFF => .SupplementaryPrivateUseAreaA,
+        0x100000...0x10FFFF => .SupplementaryPrivateUseAreaB,
+        else => error.InvalidUtf8
```

Comment:
There are still unassigned blocks; they won't be invalid in future.

Also the error is no utf8 specific, better to use `error.InvalidCodepoint` or something.

Reply:
Yeah that probably would be a better error